src/smtp/SMTPClient.ts

```diff
@@ -34,6 +34,18 @@ export default class SMTPClient {
 		})
 	}
 
+	read() {
+		return new Promise<string>(resolve => {
+			this.socket.once("data", (recv: Buffer) => {
```

Comment:
Why are you using once() ?
There could be data received between one data event and the next read() call, which is subsequently dropped.

I'd recommend listening using on() and saving the incoming data in an array. Calling read() will either take the first element out of that array (.shift()) or wait for data.

Reply:
The function is only useful right after opening the socket, so doesn't make much sense

Reply:
hmm idk feels wrong but whatever